arangod/Graph/GraphManager.cpp

```diff
@@ -554,19 +554,13 @@ Result GraphManager::ensureCollections(Graph* graph, bool waitForSync) const {
     } else if (!res.is(TRI_ERROR_ARANGO_DATA_SOURCE_NOT_FOUND)) {
       return res;
     } else {
-      if (edgeCollectionsToCreate.find(vertexColl) == edgeCollectionsToCreate.end()) {
-        if (graph->isHybrid()) {
-          if (satelliteDocumentCollectionsToCreate.find(vertexColl) ==
-                  satelliteDocumentCollectionsToCreate.end() &&
-              graph->satelliteCollections().find(vertexColl) !=
-                  graph->satelliteCollections().end()) {
-            satelliteDocumentCollectionsToCreate.emplace(vertexColl);
-          } else {
-            documentCollectionsToCreate.emplace(vertexColl);
-          }
-        } else {
-          documentCollectionsToCreate.emplace(vertexColl);
-        }
+      if (graph->isHybrid() &&
```

Comment:
I think this is now broken.
You merged 3 ifs not 2.

The behavior should be the following:
if vertexCol is in edgeCollectionsToCreate => Do nothing
else if hybrid() and vertexCol is in satelliteCollections => emplace in satelliteToCreate
else emplace in documentsToCreate

Reply:
Thanks for spotting, done.